main.tf

```diff
@@ -180,26 +192,27 @@ data "aws_iam_policy_document" "logging" {
   }
 }
 
-resource "aws_iam_policy" "default" {
-  count = local.enabled ? 1 : 0
+resource "aws_iam_policy" "s3_access_for_sftp_users" {
+  for_each = local.enabled ? local.user_names_map : {}
 
-  name   = module.iam_label.id
-  policy = join("", data.aws_iam_policy_document.allows_s3[*].json)
+  name   = "${module.iam_label.id}-${each.value}"
```

Comment:
Shouldn't this use a separate label module reference in case someone changes the delimiter?

Reply:
Yup! oops

Reply:
It would also be OK to replace the `-` with `${module.iam_label.delimiter}`, but a new label module is better because it will also handle case conversion and forbidden character removal on `each.value`.